public/components/trace_analytics/components/common/redirection_helpers.tsx

```diff
@@ -0,0 +1,183 @@
+  fromTime,
+  toTime,
+  dataSourceMDSId,
+  spanId,
+}: {
+  fromTime: string;
+  toTime: string;
+  dataSourceMDSId: Array<{ id: string; label: string }>;
+  spanId: string;
+}) => {
+  const correlatedSpanField = TraceSettings.getCorrelatedLogsFieldMappings().spanId;
+  // For telemetry lag, moving time range to +-30 minutes of startTime and endTime
+  const startTime =
+    moment(fromTime).subtract(30, 'minutes').format(TRACE_ANALYTICS_DATE_FORMAT) ?? 'now-30m';
+  const endTime = moment(toTime).add(30, 'minutes').format(TRACE_ANALYTICS_DATE_FORMAT) ?? 'now';
+  redirectionToLogsApp({
+    startTime,
+    endTime,
+    dataSourceMDSId,
+    correlatedFieldName: correlatedSpanField,
+    correlatedFieldValue: spanId,
+  });
+};
+
+export const redirectTraceToLogs = ({
+  fromTime,
+  toTime,
+  traceId,
+  dataSourceMDSId,
+}: {
+  fromTime: string;
+  toTime: string;
+  traceId: string;
+  dataSourceMDSId: DataSourceOption[];
+}) => {
+  const correlatedTraceIdField = TraceSettings.getCorrelatedLogsFieldMappings().traceId;
+  // For telemetry lag, moving time range to +-30 minutes of startTime and endTime
+  const startTime =
+    moment(fromTime).subtract(30, 'minutes').format(TRACE_ANALYTICS_DATE_FORMAT) ?? 'now-30m';
+  const endTime = moment(toTime).add(30, 'minutes').format(TRACE_ANALYTICS_DATE_FORMAT) ?? 'now';
+  redirectionToLogsApp({
+    startTime,
+    endTime,
+    dataSourceMDSId,
+    correlatedFieldName: correlatedTraceIdField,
+    correlatedFieldValue: traceId,
+  });
+};
+
+export const redirectToServiceTraces = ({
+  mode,
+  addFilter,
+  dataSourceMDSId,
+  serviceName,
+  setRedirect,
+  setCurrentSelectedService,
+}: {
+  mode: TraceAnalyticsMode;
+  addFilter: (filter: FilterType) => void;
+  dataSourceMDSId: DataSourceOption[];
+  serviceName: string;
+  setRedirect: (value: React.SetStateAction<boolean>) => void;
+  setCurrentSelectedService?: React.Dispatch<React.SetStateAction<string>>;
+}) => {
+  const newNavigation = coreRefs.chrome?.navGroup.getNavGroupEnabled();
+  if (setCurrentSelectedService) setCurrentSelectedService('');
+  setRedirect(true);
+  const filterField =
+    mode === 'data_prepper' || mode === 'custom_data_prepper'
+      ? 'serviceName'
+      : 'process.serviceName';
+  addFilter({
+    field: filterField,
+    operator: 'is',
+    value: serviceName,
+    inverted: false,
+    disabled: false,
+  });
+
+  const tracesPath = '#/traces';
+  const dataSourceId = dataSourceMDSId[0]?.id || '';
+  const urlParts = window.location.href.split('?');
+  const queryParams =
+    urlParts.length > 1 ? new URLSearchParams(urlParts[1]) : new URLSearchParams();
+
+  const modeParam = queryParams.get('mode') || '';
+  const modeQuery = modeParam ? `&mode=${encodeURIComponent(modeParam)}` : '';
+
+  if (newNavigation) {
+    coreRefs.application?.navigateToApp(observabilityTracesNewNavID, {
+      path: `${tracesPath}?datasourceId=${encodeURIComponent(dataSourceId)}${modeQuery}`,
```

Comment:
is it ok to drop all other query params except `mode`?

Reply:
Yes it is. The other params are traceId and serviceName. These shouldn't be added during re-direction as those are page specific params.